r/PACKAGING.md

```diff
@@ -0,0 +1,124 @@
+
+Make draft pull requests into the [autobrew](https://github.com/autobrew) and
+[rtools-packages](https://github.com/r-windows/rtools-packages) repositories
+used by the configure script on MacOS and Windows:
+
+- [ ] Open a draft pull request to modify 
+  [the apache-arrow autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow.rb) 
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb)
+- [ ] Open a draft pull request to modify 
+  [the apache-arrow-static autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow-static.rb) 
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow-static.rb)
+- [ ] Open a draft pull request to modify the 
+  [autobrew script](https://github.com/autobrew/scripts/blob/master/apache-arrow)
+  to include any additions made to
+  [r/tools/autobrew](https://github.com/apache/arrow/blob/master/r/tools/autobrew).
+- [ ] Open a draft pull request to modify the
+  [PKGBUILD script](https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-arrow/PKGBUILD)
+  to reflect changes in
+  [ci/PKGBUILD](https://github.com/apache/arrow/blob/master/ci/scripts/PKGBUILD),
+  uncommenting the line that says "uncomment to test the rc".
+
+Prepare and check the .tar.gz that will be released to CRAN:
+
+TODO: The packaging guide says `make release` but isn't there also a step where
+we bundle Arrow C++ into tools/cpp and prune it so that the .tar.gz stays under 5 MB?
+
+- `git fetch upstream && git checkout release-X.X.X-rc0`
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on Linux
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on Windows
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on MacOS
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on MacOS M1
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on Linux
+- [ ] `rhub::check("arrow_X.X.X.tar.gz", platform=c("debian-gcc-patched", "fedora-clang-devel"))`
+  (TODO: as I understand it, we can't do rhub Windows and Mac checks because those
+  would require the PRs into autobrew/rtools-packages to be finalized? Or maybe
+  these already know about/use the nightly binaries?)
+- [ ] Run reverse dependency checks locally (TODO: document how to do this
+  since we can't just use `revdepcheck`)
+
+After this step, we need an official release:
+  
+- [ ] Wait for the release vote to pass
+
+TODO is this where we create a CRAN release branch where the minor CRAN-specific
+fixes get picked from master? What is the git incantation for that?
```

Comment:
This was usually done in a fork afaik but I think we should not do that if possible. One of the things that always happened was removing the badges from the README as the links were sometimes flakey and causing issues. We should just add that (should be doable with a bit of `sed`) to `make release`